File: app/src/main/java/com/star_zero/example/androidmvvm/domain/task/TaskRepository.java
package com.star_zero.example.androidmvvm.domain.task;

import java.util.List;

import rx.Observable;

public interface TaskRepository {

    TaskId generateTaskId();

    Observable<Boolean> save(Task task);

    Observable<Boolean> update(Task task);

    Observable<Boolean> delete(Task task);

    Observable<List<Task>> getTasks();

}

File: app/src/main/kotlin/com/star_zero/example/androidmvvm/domain/task/TaskRepository.kt
package com.star_zero.example.androidmvvm.domain.task

import com.star_zero.example.androidmvvm.utils.AsyncLiveData

interface TaskRepository {

    fun generateTaskId(): TaskId

    fun save(task: Task): AsyncLiveData<Unit>

    fun update(task: Task): AsyncLiveData<Unit>

    fun delete(task: Task): AsyncLiveData<Unit>

    fun fetchTasks(): AsyncLiveData<List<Task>>
}